galsim/chromatic.py

```diff
@@ -131,6 +136,216 @@ def __init__(self, obj):
+                                whichever wavelength has the highest Nyquist frequency.
+                                `oversample_fac`>1 results in higher accuracy but costlier
+                                pre-computations (more memory and time). [default: 1]
+        """
+        # Set up the interpolation (which can be a costly step, depending on the length of
+        # `waves`).
+        self.waves = np.sort(np.array(waves))
+
+        # Make the objects between which we are going to interpolate.  Note that these do not have
+        # to be saved for later, unlike the images.
+        objs = [ self.evaluateAtWavelength(wave) for wave in waves ]
+
+        # The evaluation step (above) led to the incorporation of any shears and other
+        # transformations into the stored models between which we interpolation, so reset the
+        # internal attributes that store information about transformations, if indeed there were any
+        # transformations.
+        if hasattr(self, '_A') and not all ([self._nullTransformation(w) for w in waves]):
+            # Store the old transformations.
+            self._save_A = self._A
+            self._save_fluxFactor = self._fluxFactor
+            # Then reset the one that is stored as part of this profile.
+            self._A = lambda w: np.matrix(np.identity(3), dtype=float)
+            self._fluxFactor = lambda w: 1.0
+
+        # Check the fluxes for the objects.  If they are unity (within some tolerance) then that
+        # makes things simple.  If they are not, however, then we have to reset them to unity, and
+        # modify the SED attribute, which now refers to the total flux at a given wavelength after
+        # integrating over the whole light profile.
+        fluxes = np.array([ obj.getFlux() for obj in objs ])
+        if np.any(abs(fluxes - 1.0) > 1000.*np.finfo(fluxes.dtype.type).eps):
+            # Figure out the rescaling factor for the SED.
+            rescale_fac = np.zeros_like(self.waves)
+            for ind in range(len(self.waves)):
+                rescale_fac[ind] = fluxes[ind]
+                objs[ind].setFlux(1.0)
+            if not hasattr(self, 'SED'):
+                self.SED = lambda w : 1.0
+            self.SED = galsim.LookupTable(x=self.waves, f=self.SED(self.waves)*rescale_fac,
+                                          interpolant='linear')
+
+        # Find the Nyquist scale for each, and to be safe, choose the minimum value to use for the
+        # array of images that is being stored.
+        nyquist_dx_vals = [ obj.nyquistScale() for obj in objs ]
+        dx = min(nyquist_dx_vals) / oversample_fac
+
+        # Find the suggested image size for each object given the choice of scale, and use the
+        # maximum just to be safe.
+        possible_im_sizes = [ obj.SBProfile.getGoodImageSize(dx, 1.0) for obj in objs ]
+        n_im = max(possible_im_sizes)
```

Comment:
`n_im` sounds like number of images.  Recommend `im_size`, which reads more clearly as being the max of all the possible image sizes here.